src/client.ts

```diff
@@ -608,20 +613,33 @@ export class McpdClient {
    * This helper fetches server names (if not provided) and filters to only healthy servers.
    * Used by getToolSchemas(), getPrompts(), and agentTools() to avoid timeouts on failed servers.
    *
-   * @param servers - Optional array of server names. If not provided, fetches all servers.
-   * @returns Array of healthy server names
-   * @internal
+   * If logging is enabled, warnings are logged for servers that are skipped
+   * due to unhealthy status or non-existence.
+   *
+   * @param servers - Optional list of server names to filter. If not provided,
+   *                  checks health of all servers.
+   * @returns List of server names with 'ok' health status.
    */
   async #getHealthyServers(servers?: string[]): Promise<string[]> {
```

Comment:
I would restrict the type here to always be an array, I dont think it makes sense to call `getHealthyServers` with `undefined`, passing an `[]` would work aswell without explicit `null`/`undefined` checks


```suggestion
  async #getHealthyServers(servers: string[]): Promise<string[]> {
```

Reply:
It was mainly so we could call it like:

```typescript
const healthyServers = await this.#getHealthyServers(servers);
```

or 

```typescript
const allHealthyServers = await this.#getHealthyServers();
```

I'll leave it for now, and we can change it later if needed.